plugin.audio.podcasts/addon.py

```diff
@@ -0,0 +1,327 @@
+
+        self._menu = [
+            {  # root
+                "path": "",
+                "node": groups
+            }
+        ]
+
+    def _load_opml(self, path):
+
+        try:
+            with open(path) as _opml_file:
+                _data = _opml_file.read()
+                return xmltodict.parse(_data)
+        except:
+            xbmc.log("Cannot open opml file", xbmc.LOGERROR)
+            return None
+
+    def _loadRss(self, url):
+
+        res = requests.get(url)
+        if res.status_code == 200 and res.text.startswith("<?xml"):
+            return xmltodict.parse(res.text)
+
+        elif res.status_code != 200:
+            raise HttpStatusError(
+                "Unexpected HTTP Status %i for podcast %s" % (res.status_code, url))
+
+        else:
+            raise HttpStatusError("Unexpected content for podcast %s" % url)
+
+    def _load_rss_feed(self, url):
+
+        rss_feed = self._loadRss(url)
+
+        channel = rss_feed["rss"]["channel"]
+
+        if "image" in channel and "url" in channel["image"]:
+            image = channel["image"]["url"]
+        elif "itunes:image" in channel:
+            image = channel["itunes:image"]["@href"]
+        else:
+            image = ""
+
+        items = channel["item"]
+
+        return items, image
+
+    def playRss(self, parent, path, params):
+
+        try:
+            url = params["url"][0]
+            items, image = self._load_rss_feed(url)
+            index = int(params["index"][0])
+
+            item = items[index]
+            if "enclosure" in item:
+                stream_url = item["enclosure"]["@url"]
+            else:
+                stream_url = item["guid"]
+
+            xbmc.executebuiltin('PlayMedia(%s)' % stream_url)
+
+        except HttpStatusError as error:
+
+            xbmcgui.Dialog().notification("HTTP Status Error", error.message)
+
+    def renderRss(self, parent, path, params):
+
+        url = params["url"][0]
+        try:
+            items, image = self._load_rss_feed(url)
+
+        except HttpStatusError as error:
+
+            xbmcgui.Dialog().notification("HTTP Status Error", error.message)
+
+        index = 0
+
+        entries = []
+
+        for item in items:
+
+            entries = entries + [{
+                "path": str(index),
+                "name": item["title"],
+                "name2": item["description"] if "description" in item else "",
+                "icon": image,
+                "params": [
+                    {
+                        "call": "playRss",
+                        "index": str(index),
+                        "url": url
+                    }
+                ]
+            }]
+
+            index += 1
+
+        for entry in entries:
+
+            self._add_list_item(entry, path)
+
+        xbmcplugin.endOfDirectory(self._addon_handle)
+
+    def play(self, parent, path, params):
+
+        url = params["url"][0]
+        xbmc.executebuiltin('PlayMedia(%s)' % url)
```

Comment:
I checked and a plugin should use xbmcplugin.setResolvedUrl() in combination with xbmcgui.ListItem.setProperty('IsPlayable', 'true'). That way Kodi will also remember what items were played in the past. But I guess this works. Same in other places.

Reply:
@basrieter I'm going to investigate what setResolvedUrl does and if it fits into my approach.

So far I have read this article: [What is the goal of the method "setResolvedUrl" ?](https://forum.kodi.tv/showthread.php?tid=173986)

Actually the URLs inside the plugin are like pointers to items in RSS feed, e.g. ˋˋˋpod-1/1ˋˋˋ which means first RSS feed and first entry in this feed. The advantage is that in this example the URL always points to the most recent podcast and therefore not to a speicific one. This is helpful when bookmarking it or using it in combination with my [timer addon](https://github.com/Heckie75/kodi-addon-heckies-timer) which I maybe want to post later.

As mentioned I will come back to you after further investigation and after I know in detail how setResolvedUrl works.